check_correctness/src/mesh.cpp

```diff
@@ -0,0 +1,64 @@
+#include "mesh.h"
+
+#include <mpi.h>
+
+#include "PUML/PUML.h"
+#include "PUML/Neighbor.h"
+#include "PUML/Downward.h"
+
+Mesh::Mesh(const std::string& fileName) {
+  puml.open((fileName + ":/connect").c_str(), (fileName + ":/geometry").c_str());
+  puml.addData((fileName + ":/group").c_str(), PUML::CELL);
+  puml.addData((fileName + ":/boundary").c_str(), PUML::CELL);
+  int numPartitionCells = puml.numOriginalCells();
+  int rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  // Keep all cells on the same rank
```

Comment:
so you read the full mesh on all ranks?

Reply:
No, but I don't do a partitioning. Rank X reads some cells and keeps exactly these cells.